### DISCUSSION TITLE
Where clause on Include type

### DISCUSSION THREAD

**Discussion by KevinRecuerda** [JasperFx/marten#1357] [Ideas]
Hello,

I would like to filter on joined table:
```csharp
User included = null;
var issues = query.Query<Issue>()
                  .Include<User>(x => x.AssigneeId, x => included = x)
                  .Where<User>(x => x.UserName == "toto")
                  .ToList();
```

Any idea to achieve this ?

**Comment by mysticmind** [comment]
@KevinRecuerda With current available functionality, you can only do a filter on the queried document. And `Include` only brings the related documents. Possibly, one way is to solve your use case is to do a local filter after fetching of the records.

We could look at providing functionality to filter on the `Include` document type in future.

**Comment by KevinRecuerda** [comment]
Local filter works, but it's not efficient. Looking my example, we will get all data from tables.

In my mind, this could be a great functionality.

Moreover, i created an extension method to solve it.
https://github.com/KevinRecuerda/DocumentStore/blob/master/DocumentStore/Extensions/MartenQueryableExtensions.cs#L59

**Comment by jeremydmiller** [comment]
@KevinRecuerda @mysticmind It's not a crazy feature to add. Might not be that bad to implement. Let's see how things are going if this goes into v4 or slips to 4.* though.

**Reply by RKopczynski** [reply]
I just stumbled upon the same problem. Are any moves to implement this? Seems like very useful thing.

**Reply by jeremydmiller** [reply]
Nothing started yet, no. We're struggling to bring the existing issues in for V4 as is, so it's just a bandwidth issue.

**Reply by Gafs63** [reply]
Hello,
Is this feature now available in v4 or V5 ?
